### PR TITLE
Feature/17 lib factories

### PR DESCRIPTION
**Pull request by Cavonstavant**

### CODE REVIEW

**Review by pariciadra — APPROVED** [review]

File: Src/Core/LibManager.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        _libsHandle.emplace(libPath, nullptr);
+    }
+}
+
+void *LibManager::openLib(std::string &libPath)
+{
+    auto it = _libsHandle.find(libPath);
+
+    if (it == _libsHandle.end()) {
+        LibraryEX("Library not found", Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:31]: throw

File: Src/Core/LibManager.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        exit(84);
+    }
+    if (it->second == nullptr) {
+        it->second = dlopen(libPath.c_str(), RTLD_LAZY);
+        if (it->second == nullptr) {
+            LibraryEX(dlerror(), Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:37]: throw

File: Src/Core/LibManager.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            exit(84);
+        }
+    } else {
+        LibraryEX("Library already opened", Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:41]: throw

File: Src/Core/LibManager.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    return (it->second);
+}
+
+void LibManager::closeLib(std::string &libPath)
+{
+    auto it = _libsHandle.find(libPath);
+
+    if (it == _libsHandle.end()) {
+        LibraryEX("Library not found", Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:52]: throw

File: Src/Core/LibManager.cpp [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        exit(84);
+    }
+    if (it->second != nullptr) {
+        dlclose(it->second);
+        it->second = nullptr;
+    } else {
+        LibraryEX("Library already closed", Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:59]: throw

File: Src/Core/LibManager.cpp [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        exit(84);
+    }
+}
+
+void *LibManager::getSymbol(void *libHandle, std::string &symbolName)
+{
+    void *symbol = dlsym(libHandle, symbolName.c_str());
+
+    if (symbol == nullptr) {
+        LibraryEX(dlerror(), Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:69]: throw

File: Src/Core/LibManager.cpp [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    return (symbol);
+}
+
+void *LibManager::getSymbol(std::string &libPath, std::string &symbolName)
+{
+    auto it = _libsHandle.find(libPath);
+
+    if (it == _libsHandle.end()) {
+        LibraryEX("Library not found", Logger::CRITICAL);
```
Comment (aurelenc) [inline Src/Core/LibManager.cpp:80]: throw

**Review by aurelenc — CHANGES_REQUESTED** [review]

**Review by aurelenc — APPROVED** [review]

**Pull request merged**